### DISCUSSION TITLE
My odd emporia that won't boot

### DISCUSSION THREAD

**Discussion by archfolf** [emporia-vue-local/esphome#252] [Show and tell]
So, I was really excited to flash and test my Emporia device that I got off eBay, and maybe that is why my unit has this really odd problem. It won't flash! When I first got it, it powered on and seemed to do its normal beeping thing when it booted up, so I soldered all the pins and got ready for flashing.

![24-01-06 21-27-32 0565](https://github.com/emporia-vue-local/esphome/assets/33212687/bf4cc88e-881a-46d4-928f-2de27c71a2a4)

That didn't go well. I thought that maybe my understanding of how to flash things with a Raspberry Pi was not so good, or maybe the Pi is just not really good at this TTL thing because I got nothing from esptool, only a headache. So, off to Amazon I went, and a few days later, I got myself a real USB To TTL thing. I even flashed an ESP32 I had lying around just to be sure it worked! But before that, I tried to flash the Emporia with the same ESP32 I had lying around by holding down the EN button on that ESP32. I thought it worked, but I was just flashing the ESP32 I was holding the end button down; a red herring. I even took up the Emporia and everything. With that, I didn't know where else to go. It looks like the onboard ESP32 on the Emporia won't boot, won't even beep at me. It looks like it's dead, so I told the eBay seller as much, and they were going to take it back. However, they also told me to contact support, so I did, and support sent me a new one. I mean, gosh, that is some good support.

But that would take a few days. In the meantime, my USB to TTL programmer arrived, and it had RTS and CTS pins on it, so I hooked those up. When I did, the Emporia beeped at me, but only sometimes. I don't know what was going on. Debugging electronic devices is not something I do. I just thought that maybe it would restart somehow. I got some reading from it before; I got nothing when trying to back up the flash with esptool. Now, at least, I am getting ESP32: INVALID HEAD OF PACKET (0X00). So, I am getting somewhere. After doing things like switching TX and RX around, even sending 5V to 3.3V, I still don't get much further than that. After thinking this thing is dead anyway, I might as well try sending 3.3V down every pin, starting with 3.3V, and when I got to EN, "beep," and again "beep," it worked. It worked! I had no clue why. I guess something went wrong with the EN circuitry on this Emporia, so the ESP32 was not dead; it was just being held in reset mode.

![24-01-13 13-41-57 0568](https://github.com/emporia-vue-local/esphome/assets/33212687/828ee8d1-9bef-46b5-b3e1-2a5e6fc94e1b)

With that, I can flash! 

![24-01-13 14-16-22 0569](https://github.com/emporia-vue-local/esphome/assets/33212687/cc398669-19e8-4af8-ab57-b3b76eb799c3)

Here is my fix: by drawing power from the 3.3V pin and directing it to the EN pin. And it works! Until I had to reflash it because I tried to add Bluetooth proxy to it. Yeah, don't do that. Whatever broke that circuit, broke it even worse. My trick of sending 3.3V to the EN pin didn't work, so I had to break the two resistors in the back of the unit off that were in cicuit with the EN pin, and then it just flashed with the EN power injection just fine. I now have the replacement unit from Emporia themselves. so, if anything happen. I can just replace it.

I'm just happy that I could, and I guess, fix something.

TL;DR: EN circuitry broken, had to inject power from the 3.3V pin to get it to work.

**Comment by Darkyputz** [comment]
Dude, on your first picture, only apply 5 OR 3.3 volts, not both.
I have the same USB flasher and i can only flash it and run it with en and gpio0 attached, other than that it is not booting.
So if you find a solution, hand it down

**Reply by chrisgilldc** [reply]
Just wanted to note that I banged my head for a couple hours to find a way to flash my Vue 2 (Rev 4) until I came across this post. Seems for whatever reason it only likes 3.3v, not 5v.

**Reply by archfolf** [reply]
Leave it attached
